### ISSUE TITLE
Generic type parameters not resolved properly

### ISSUE THREAD

**Issue by Mobe91** [vojtechhabarta/typescript-generator#380]
Given to following class hierarchy:

```java
public interface IdView<T extends Serializable> {
    T getId();
}

public interface Entity1IdView extends IdView<UUID> {
}

public abstract class Entity1View implements Entity1IdView {
  private String name;

  public String getName() { return name; }
}
```

Yields the following typescript definitions:

```typescript
export interface Serializable {
}
export interface Entity1IdView extends IdView<string> {
}
export class Entity1View implements Entity1IdView {
    id: T;
    name: string;
}
```

Apparently, the type parameter `T` is not resolved correctly for the declaration of the `id` field in `Entity1View`.

**Comment by olegshtch** [comment]
`cz.habarta.typescript.generator.compiler.ModelCompiler.getTypeParameters` ignores bounds. But it possible to have unknown class here, so it should use data from `Model`, which discovers new classes.

**Comment by olegshtch** [comment]
@Mobe91 Could you try #389 ?

**Comment by vojtechhabarta** [comment]
@olegshtch I tried #389 with mentioned example, here is the generated output:

``` typescript
export class Entity1View implements Entity1IdView {
    id: T;  // needs to be resolved
    name: string;
}
export interface Entity1IdView extends IdView<string> {
}
export interface IdView<T extends Serializable> {  // added generic bound
    id: T;
}
export interface Serializable {
}
```

I think it is needed to resolve generic variable `T` in `Entity1View` to `string`. Resolve generic variables is relatively complex in general case. I am thinking whether to use some library for it or implement it in typescript-generator.

This would be also useful for #219.

**Comment by olegshtch** [comment]
It's strange but after rebase https://github.com/vojtechhabarta/typescript-generator/pull/389 on top `master` I get 

```typescript
interface Entity1View extends Entity1IdView {
    name: string;
}
interface Entity1IdView extends IdView<string> {
}
interface IdView<T extends string> {
    id: T;
}
```
from:
```java
    public interface IdView<T extends String> {
        T getId();
    }

    public interface Entity1IdView extends IdView<String> {
    }

    public abstract class Entity1View implements Entity1IdView {
        private String name;

        public String getName() { return name; }
    }
```
It looks correct.

**Comment by olegshtch** [comment]
Looks like somehow it should generate `class` instead of `interface` to reproduce bug.

**Comment by vojtechhabarta** [comment]
Fix released in version [2.17.558](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.17.558).
